### ISSUE TITLE
Half suggestion / half potential logical error: Using 3 inputs

### ISSUE THREAD

**Issue by HanatakeYurii** [tristankechlo/AdditionalRedstone#4]
First of all, the way you take your mod is very interesting, it's the first time I see a mod using 3 inputs for logic gates. I think that's a pretty clever idea given how Minecraft works. But at the same time there are some unused potential, and I think I may have found a logical error on your wiki. 

Since there is 3 inputs, there exist 14 gates instead of the classic 6 for 2 inputs (((2^number of inputs)-1)*2). Those gates have not common names (but there are multiple ways to represent them), so I'll try to be clear:
- full AND ("&" or "=3") that requires all inputs to be powered (as you did);
- full XOR ("=1") that requires exactly 1 powered input (not what you did, which is a bit strange);
- full OR ("≥1") that requires at least 1 powered input (as you did);
- partial AND ("≥2") that requires at least 2 powered inputs;
- partial XOR ("1||2") that requires 1 to 2 powered inputs but not 3;
- partial XOR ("1||3") that requires either 1 or 3 powered inputs (that's the one you strangely called XOR);
- partial XOR ("=2") that requires exactly 2 powered inputs;
- _and you add to that all 7 reversed, thus making 14 gates._
You might consider completing the set, using symbols as you did before to make it more clear.

At the moment that's all I come with while keeping it simple. I might have another ideas but I'm not sure if wouldn't be a bother code-wise or even performant in-game.

Thanks for your time

**Comment by tristankechlo** [comment]
Thank you for your ideas/suggestions, it's always nice when people participate and want to improve things, so if you have more ideas let me know  
  
**regarding my implementation of the XOR:**  
I am using 'odd parity' to determine the output (uneven amount of inputs) since this is the most commonly accepted version.  
Yes, the symbol `=1` on the gate might be a bit misleading, but I couldn't figure out a neat symbol to fit on the block ;D  
  
**your other gate suggestions:**  
Interesting ideas, I never came across them before, but I might implement them in the future.  
Shouldn't be there a partial XOR with 2||3 too?

**Comment by HanatakeYurii** [comment]
"2||3" would be the same as "≥2" since both allows 2 or 3 powered inputs. But other than that asymmetric gates would also be possible (like with some inputs having precedence or something like 1 AND 2 OR 2 AND 3 but not 1 AND 3), but those cases are rare and can be done by combining multiple symmetric gates so I wouldn't recommending doing those (in fact those are even possible for 2 inputs but it's no surprise that nobody do them).

**Comment by tristankechlo** [comment]
oh its meant to read any 1 or 2 inputs active, and not input no.1 and input no.2 active  
that makes more sense

**Comment by HanatakeYurii** [comment]
Yeah, that's the count number of powered inputs, not hypothetical input's designated number (that would be an asymmetric gate).

PS: I understood what you meant for the 3-input XNOR truth table, there is a result difference between natural language (what I expected) and associativity (the result you made).

**Comment by ColonelGerdauf** [comment]
I just noticed this, and I will offer a bit of my own feedback:

- "1||2"; isn't that better written as "≤2"?
- Isn't the partial AND better stated as an incomplete XNOR?

**Comment by tristankechlo** [comment]
Small update, I decided not to add those extra gates.  
But with the future version 1.2.0 there will be a `supergate`, which can be customized to whatever logic players like.  
The gates already implemented by the mod will stay for convenience.

**Comment by HanatakeYurii** [comment]
Woah, that's awesome, I imagined something like that for a time but I didn't get how difficult it would have been. Glad you did.
In the end, it's one gate to rule them all.

**Comment by tristankechlo** [comment]
This is now done for all versions up to 1.21.1.  
Closing this for now
